src/instructions.py

```diff
@@ -249,15 +249,33 @@ def decode_consume_events(instruction: TransactionInstruction) -> ConsumeEventsP
 
 
 def decode_cancel_order(instruction: TransactionInstruction) -> CancelOrderParams:
-    raise NotImplementedError("decode_cancel_order not implemented")
+    validate_instruction_keys(instruction, 4)
+    data = INSTRUCTIONS_LAYOUT.parse(instruction.data)
```

Comment:
Can we condense all the decode and validation logic to something like https://github.com/michaelhly/solana-py/blob/682f0f67015842b60b0edec5ca4e6b25d731e93c/spl/token/instructions.py#L291

Reply:
Good idea. This is done.